cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -769,17 +782,26 @@ private WorkflowProgramAdder addProgram(String phaseName, WorkflowProgramAdder p
     return programAdder;
   }
 
-  private WorkflowProgramAdder addConditionBranches(WorkflowProgramAdder conditionAdder, ConditionBranches branches) {
+  private WorkflowProgramAdder addConditionBranches(WorkflowProgramAdder conditionAdder, BatchPipelineSpec pipelineSpec,
+                                                    ConditionBranches branches) {
     // Add all phases on the true branch here
     String trueOutput = branches.getTrueOutput();
     if (trueOutput != null) {
-      addPrograms(trueOutput, conditionAdder);
+      addPrograms(trueOutput, conditionAdder, pipelineSpec);
     }
     String falseOutput = branches.getFalseOutput();
     if (falseOutput != null) {
       conditionAdder.otherwise();
-      addPrograms(falseOutput, conditionAdder);
+      addPrograms(falseOutput, conditionAdder, pipelineSpec);
     }
     return conditionAdder.end();
   }
+
+  private BatchPipelineSpec getPipelineSpec() {
```

Comment:
I think it's better not to have this method, since it can't actually be called anywhere except in configurePipeline. It will null pointer exception if called from anywhere else, as the config variable won't be set.

Reply:
Done.